spec/blocksync/verification.md

```diff
@@ -0,0 +1,120 @@
+## Block verification
+
+When blocksyncing, a node is not participating in conensus. It is receiving blocks that have already been decided and committed. To avoid being fooled by malicious peers, the node has to verify the received blocks before executing the transactions and storing the block in its store. 
+
+
+The verification in blocksync aims to apply the same logic as the [light client verification](../light-client/verification/README.md). The safety guarantees of the light client model are provided by verifiying blocks starting from an **initial trusted state** and using validators who were bonded within a **trusting period**. Once this period expires, the validators could have unbonded, and we cannot rely on them being hoenst anymore. In blocksync, we can be **outside** the trusting period. Therefore, the guarantees provided by blocksync verification can be divided in two groups: 
+
+- The block is within the trusting period, we can punish validators, and provide guarantees that if a block is signed by trusted validators, they are indeed correct. 
+- The block is outside the trusting period and we cannot guarantee that the validators are still honest. We can however make a potential attack by those validators more complicated by additionally verifying the block using other peers as *witnesses*.
+
+A state becomes trusted once it passes the validation. The section below discusses this state in more details.
+
+### Trusted state
+
+Currently there are two possible ways to obtain a trusted state :
+ 1. If the node is blocksync-ing from genesis, we have to verify the very first block and mark it as the initial trusted state.
+ 
+ [//]: # (trust the validators provided in the initial state and use them to verify the initial block received from peers.)
+
+ 2. The node loads the last block stored in its local block store. 
+
+**Case 1 - blocksync from genesis**
+
+In this case we only verify that the validator hash of the block matches the validator set of the initial state. We will use the initially provided validator set for verification and further verify the block against witne+sses. In this scenario, this state is very likely to be outside the trusting period. We will accept this block as a trusted state and store it inside the node's block store. 
```

Comment:
I have one thought here. I understand it is reasonable to trust the validator set from genesis (social consensus), and therefore the first block if the validators hash matches. But then, as we advance with later blocks, but still far away from the trusting window, someone could try to feed us with a fake block forking from the actual chain. How do we deal with this case?

Reply:
We witness verify, even the initial block. But this also does not provide us 100% correctness guarantees as the witnesses could be lying. But if we assume at least one is correct, we should be able to detect a divergence and realize someone is trying to fool us. The problem is how would we determine who that is.

Reply:
I see. Then I think it's important to stress somewhere/somehow that, until we reach the trusting window, we cannot be sure we are applying the right blocks. And thus, the operator should be ready to accept that, right at the end, the node is unusable and they need to start over (we should also point out that this is extremely unlikely)

Reply:
Indeed. It is also worth noting that, as a faulty node cannot forge signatures, if we assume the 2/3 correctness model, we should have some certainty that if the verification passes, the blocks are also correct.